span.go

```diff
@@ -121,6 +124,18 @@ type LogData struct {
 	Payload interface{}
 }
 
+// FinishOptions allows Span.FinishWithOptions callers to override the finish
+// timestamp and provide log data via a bulk interface.
+type FinishOptions struct {
+	// FinishTime overrides the Span's finish time, or implicitly becomes
+	// time.Now() if FinishTime.IsZero().
+	FinishTime time.Time
+
+	// BulkLogData allows the caller to specify the contents of many Log()
+	// calls with a single slice. May be nil.
+	BulkLogData []LogData
```

Comment:
Assumptions about timestamps are somewhat natural, which could trip up the changes between implementations. Let's be clear about what can be assumed on LogData and the various other timestamps. I'm ok with saying that there are no guarantees whatsoever, but there should be some shared interpretation of a span that starts tomorrow, ends yesterday and has event timestamps from 23BC to 2142.

Also, unfortunately (at least with Go) ownership needs clarification: Is the slice off-limits for the caller after having been passed in?